### DISCUSSION TITLE
sourceUrl is taken from another package with the same name

### DISCUSSION THREAD

**Discussion by clementfumey** [renovatebot/renovate#24570] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.93.7 docker image renovate/renovate:36-full

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I am using renovate from Jenkins and Docker to update dependencies of a python package.

In this python package, dependencies are handled by PDM and/or Poetry (in separate test branches, see : [this renovate discussion](https://github.com/renovatebot/renovate/discussions/24436) ), there is both pypi.org and a private nexus pypi for some private packages. 

When renovate run, upgrade for a private package named `ota` is correctly handled on the private pypi but the sourceUrl is taken from a package named `ota` in pypi.org ("sourceUrl": "https://github.com/royaurelien/ota-client",) which has nothing to do with ou package. Our private `ota` package has a `Project-URL:` set up. This happens both with the pdm and poetry manager and also with another package named obderver

In my case sourceUrl is needed to make a packageRules with match sourceUrlPrefix off all our private packages. (A matchRegistryUrl config could have been convenient by the way)

I am not able to provide a minimal reproduction repository has our private pypi is not publicly accessible

### Relevant debug logs

<details><summary>Logs</summary>

```
               {
                 "packageName": "ota",
                 "depName": "ota",
                 "datasource": "pypi",
                 "depType": "project.dependencies",
                 "currentValue": "==1.0.15",
                 "registryUrls": [
                   "https://nexus***.*******.com/repository/pypi/simple/",
                   "https://pypi.org/simple"
                 ],
                 "updates": [],
                 "versioning": "pep440",
                 "warnings": [],
                 "sourceUrl": "https://github.com/royaurelien/ota-client",
                 "currentVersion": "1.0.15",
                 "fixedVersion": "1.0.15"
               },
               {
                 "packageName": "observer",
                 "depName": "observer",
                 "datasource": "pypi",
                 "depType": "project.dependencies",
                 "currentValue": "==1.0.4",
                 "registryUrls": [
                   "https://nexus***.*******.com/repository/pypi/simple/",
                   "https://pypi.org/simple"
                 ],
                 "updates": [],
                 "versioning": "pep440",
                 "warnings": [],
                 "sourceUrl": "https://github.com/htmue/python-observer",
                 "currentVersion": "1.0.4",
                 "fixedVersion": "1.0.4"
               },
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by secustor** [comment]
Results are merged for pypi for different datasources. As there is no sourceUrl in your private registry it pulls it from the public one.

**Reply by clementfumey** [reply]
There is no source url in the public package either. I guess renovate take it from the Project-URL: or the Home-page: metadata of the package but I see no reason for renovate to take metadata from the public package if it uses the private one.

**Comment by rarkins** [comment]
One question is: does Renovate detect the source URL from your private registry in the first place? You wrote:

> Our private ota package has a Project-URL: set up

But I don't know if that means it's detected. If not, then ignoring the public URL won't help
